src/main/java/lotto/domain/Prize.java

```diff
@@ -23,17 +24,23 @@
 		this.reward = reward;
 	}
 
-	public static Prize of(Long count) {
-		return Arrays.stream(values()).filter(one -> one.matchCount == count)
+	public static Prize of(Long count, boolean isMatchBonus) {
+		return Arrays.stream(values()).filter(prize -> prize.isEqualMatchCount(count))
+			.filter(prize -> !prize.equals(SECOND) || isMatchBonus)
```

Comment:
Stream은 한 줄에 하나의 처리만 하도록 구현하면 가독성을 향상시킬 수 있습니다.

```suggestion
		return Arrays.stream(values())
		        .filter(prize -> prize.isEqualMatchCount(count))
			.filter(prize -> !prize.equals(SECOND) || isMatchBonus)
```